identity/legacy.go

```diff
@@ -18,18 +19,18 @@ func (l *Legacy) Auth(creds *Credentials) (*AuthDetails, error) {
 	}
 	request, err := http.NewRequest("GET", creds.URL, nil)
 	if err != nil {
-		return nil, err
+		return nil, gooseerrors.Newf(err, "", "GET request failed")
 	}
 	request.Header.Set("X-Auth-User", creds.User)
 	request.Header.Set("X-Auth-Key", creds.Secrets)
 	response, err := l.client.Do(request)
 	if err != nil {
-		return nil, err
+		return nil, gooseerrors.NewUnauthorisedf(err, "", "request failed (maybe client policy or network connectivity issue)")
```

Comment:
Why is this Unauthorised error? The error message acknowledges that it is probably something else.
You just need to revert this to the original code.